beets/dbcore/query.py

```diff
@@ -227,6 +229,63 @@ def string_match(cls, pattern, value):
         return pattern.search(cls._normalize(value)) is not None
 
 
+class JSonSubstringListQuery(SubstringQuery):
+    """A query that matches a regular expression in a specific item
+    field which contains a json dump of a string array.
+    """
+    _unique_id = 1
+
+    def __init__(self, field, pattern, fast, model_cls):
+        super(JSonSubstringListQuery, self).__init__(field, pattern, fast)
+        self.model_cls = model_cls
+
+    def get_unique_alias(self):
+        alias = 'json_' + self.field + str(JSonSubstringListQuery._unique_id)
+        JSonSubstringListQuery._unique_id += 1
+        return alias
+
+    def col_clause(self):
+        pattern = (self.pattern
+                       .replace('\\', '\\\\')
+                       .replace('%', '\\%')
+                       .replace('_', '\\_'))
+        search = '%' + pattern + '%'
+        alias = self.get_unique_alias()
+        clause = alias + ".value like ? escape '\\'"
+        subvals = [search]
+        table = 'json_each({0}) as {1}'.format(self.field, alias)
+        return clause, subvals, (table,)
+
+    def negated_clause(self):
+        clause, subvals, table = self.col_clause()
+        _table = self.model_cls._table
+        primary_key = '%s.id' % (_table)
```

Comment:
@sampsyo Do you think it's worth it to add a get_primary_key_field to Model so instead of hardcoding `id` I call self.model_cls.get_primary_key_field() which would iterate over its fields and search which has a PRIMARY_ID type? I think that would be more elegant, but I'm not sure if it's worth the effort.